### ISSUE TITLE
Feature request: new check for `super().somefunc(self, ...)`

### ISSUE THREAD

**Issue by sirosen** [PyCQA/flake8-bugbear#168]
I looked for a request for this check in pycodestyle, pyflakes, and flake8 and didn't find one. Upon consideration, this is a better feature for flake8-bugbear than any of those projects because it would synergize so well with `B902`.

Consider this example:
```python
class Foo(Bar):
    def __init__(self, message, **kwargs):
        self.baz = 1
        super().__init__(self, message, **kwargs)
```

The mistake is obvious when isolated like this -- `self` should not be passed because `super()` implicitly passes it.
There are possibly very niche cases in which this is legitimate and correct, but, most likely, if someone wrote this it's because they slipped up.

Because `self` (and `cls`, if we include classmethods) are special names, and `super` is a keyword, it should be possible to detect this usage in a flake8 run.

The bigger question is, "Is it too trivial and rare of a mistake to be worth a check?"
I don't know how to decide that. I have forgotten exactly what the signature of `super()` is at least a couple of times. Embarrassing? Yes. But I doubt I'm alone.

If it would be a welcome contribution, perhaps as another `B9xx` check, I'd be happy to try to write this. However, I'm not sure how to do so. Since `self` isn't an argument to `super()`, I don't think I can start at the `super` node in the AST. If the idea gets a 👍 , I would appreciate any direction towards a similar check which could serve as an example.

**Comment by cooperlees** [comment]
Hi,

Thanks. Would love to discuss this and see if anyone else shares your want for this check. 

I've never personally done this and feel typing + unit tests should catch this. I get your merit of the lint and it stopping people doing It early in development tho. I'm not super strong either way here, but due to my heavy use of typing and unit tests don't see the benefit personally.

As for implementation, I'd recommend adding some code doing in a dedicated unittest fixture file and see how easy it is to do and discuss your findings here. Feel free to share a Work In progress PR to talk on too.

Let's see if anyone else weighs in and adds around this.

**Comment by sirosen** [comment]
+1 to the idea that typing and testing should catch this. This is just about catching it faster.

Maybe a week is too short, but no one else has shown up to upvote or enthusiastically endorse this.
I think I'm alone in not being able to remember `super`'s syntax! 😅

I think I should close this. If there's interest in the future, it can be reopened and I'd be happy to work on it.
